### ISSUE TITLE
XAML Documentation

### ISSUE THREAD

**Issue by brminnick** [brminnick/EntryCustomReturnPlugin#7]
Add XAML Documentation to the README

**Comment by bruce-willis** [comment]
@brminnick Is it possible to use effects throw XAML now?

**Comment by brminnick** [comment]
Yup!

https://developer.xamarin.com/guides/xamarin-forms/application-fundamentals/effects/creating/#Consuming_the_Effect_in_XAML

**Comment by brminnick** [comment]
EffectsPage.xaml
```
 <Entry
            x:Name = "EffectsEntry"
            HorizontalOptions="Center"
            VerticalOptions="Center"
            Placeholder="{x:Static simpleSampleShared:EffectsPageConstants.EffectsEntryPlaceholderText}"
            AutomationId="{x:Static simpleSampleShared:AutomationIdConstants.EffectsEntry}"
            entryCustomReturn:CustomReturnEffect.TempReturnType="{x:Static entryCustomReturn:ReturnType.Go}"/>
```

CustomReturnEffect.cs
```
 public static BindableProperty TempReturnTypeProperty= BindableProperty.CreateAttached(propertyName: "TempReturnType",
                                                                    returnType: typeof(ReturnType),
                                                                    declaringType: typeof(Entry),
                                                                    defaultValue: ReturnType.Default,
                                                                    propertyChanged: OnReturnTypeChanged);

        public static ReturnType GetTempReturnType(BindableObject view) => (ReturnType)view.GetValue(TempReturnTypeProperty);
        public static void SetTempReturnType(BindableObject view, ReturnType value) => view.SetValue(TempReturnTypeProperty, value);
```

**Comment by brminnick** [comment]
@bruce-willis I'm working on building a XAML sample in this branch: https://github.com/brminnick/EntryCustomReturnPlugin/tree/Add-Xaml-Documentation/SimpleXamlSample

I hit a snag today with how XAML handles attached properties and reflection, and I need to reach out the Xamarin.Forms team for some assistance.

For now, ignore the above comment. That was just notes from my hack session today!

**Comment by brminnick** [comment]
Here's a working example of using the Custom Renderer in XAML

``` 
<?xml version="1.0" encoding="UTF-8"?>
<ContentPage 
    xmlns="http://xamarin.com/schemas/2014/forms" 
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" 
    x:Class="SimpleXamlSample.CustomRendererPage"
    xmlns:simpleSampleShared="clr-namespace:SimpleSamples.Shared;assembly=SimpleXamlSample"
    xmlns:entryCustomReturn="clr-namespace:EntryCustomReturn.Forms.Plugin.Abstractions;assembly=EntryCustomReturn.Forms.Plugin.Abstractions"
    Title="{x:Static simpleSampleShared:PageTitles.CustomRenderer}">
    
	<ContentPage.Content>

        <entryCustomReturn:CustomReturnEntry
            x:Name = "CustomReturnEntry"
            HorizontalOptions="Center"
            VerticalOptions="Center"
            Placeholder="{x:Static simpleSampleShared:CustomRendererPageConstants.CustomReturnEntryPlaceholderText}"
            AutomationId="{x:Static simpleSampleShared:AutomationIdConstants.CustomReturnEntry}"
            ReturnType="Go"/>
        
	</ContentPage.Content>
</ContentPage>
```

**Comment by brminnick** [comment]
XAML Support for Effects is now available in v3.3.0.

v3.3.0-pre2 is available on NuGet today

**Comment by brminnick** [comment]
@bruce-willis FYI - I added XAML Support in v3.3.0, which is now live! I also updated the Readme to show how to use the plugin in XAML.
